core/application/impl/app_configuration_impl.cpp

```diff
@@ -166,14 +167,36 @@ namespace kagome::application {
 
   void AppConfigurationImpl::parse_general_segment(rapidjson::Value &val) {
     std::string roles_str;
-    load_str(val, "role", roles_str);
-    if (roles_str == "full") {
-      roles_.flags.full = 1;
-    } else if (roles_str == "light") {
-      roles_.flags.light = 1;
-    } else if (roles_str == "authority") {
-      roles_.flags.authority = 1;
+    load_str(val, "roles", roles_str);
+
+    std::vector<std::string> roles;
+    boost::split(roles, roles_str, boost::algorithm::is_punct());
+    roles_ = 0;
+    for (auto &role : roles) {
+      if (role == "light") {
+        roles_.flags.light = 1;
+        logger_->error("Role 'archive' is not supported yet");
```

Comment:
Is this error message about "archive" role redundant under the "light" case?

Reply:
```suggestion
        logger_->error("Role 'light' is not supported yet");
```
probably

Reply:
Sure